src/Features/Core/Portable/Wrapping/WrapItemsAction.cs

```diff
@@ -64,22 +63,35 @@ protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperation
         }
 
         public static ImmutableArray<CodeAction> SortActionsByMostRecentlyUsed(ImmutableArray<CodeAction> codeActions)
+            => SortActionsByMostRecentlyUsed(codeActions, s_mruTitles);
+
+        public static ImmutableArray<CodeAction> SortActionsByMostRecentlyUsed(
+            ImmutableArray<CodeAction> codeActions, ImmutableArray<string> mruTitles)
         {
-            // make a local so this array can't change out from under us.
-            var mruTitles = s_mruTitles;
-            return codeActions.Sort((d1, d2) => ComparerWithState.CompareTo(d1, d2, (mruTitles, codeActions), s_comparers));
-        }
+            return codeActions.Sort((d1, d2) =>
+            {
+                var mruIndex1 = mruTitles.IndexOf(GetSortTitle(d1));
+                var mruIndex2 = mruTitles.IndexOf(GetSortTitle(d2));
+
+                // If both are in the mru, prefer the one earlier on.
+                if (mruIndex1 >= 0 && mruIndex2 >= 0)
+                    return mruIndex1 - mruIndex2;
 
-        private static readonly ImmutableArray<Func<CodeAction, (ImmutableArray<string>, ImmutableArray<CodeAction>), IComparable>> s_comparers =
-            ImmutableArray.Create<Func<CodeAction, (ImmutableArray<string>, ImmutableArray<CodeAction>), IComparable>>(
-                // one of these has never been invoked.  It's always after an item that has been
-                // invoked.
-                // we've invoked both of these before.  Order by how recently it was invoked.
-                (ca, tuple) => tuple.Item1.IndexOf(GetSortTitle(ca)),
```

Comment:
this approach just basically does not work.  a non-invoked item will have a negative index, an invoked item will have a positive index.